docs: improve custom sandbox guide with more configuration options (#5589)
All-Hands-AI/OpenHands: docs/modules/usage/how-to/custom-sandbox-guide.md

## Create Your Docker Image

> To create a custom Docker image, you can use any base image that supports your requirements. While Debian-based images are common, they're not strictly required.

Reviewer: Did you try this with a non-debian based image? I tried it once for a simple ruby image and it did not work. Maybe it was fixed.

Author: I'll revert this...

```suggestion
To create a custom Docker image, it must be Debian based.
```

sandbox_base_container_image="custom-image"
```

### Additional Configuration Options

The `config.toml` file supports several other options for customizing your sandbox:

```toml
[core]

> # Install additional dependencies at runtime build time

Author: ```suggestion
# Install additional dependencies when the runtime is built
```

# Can contain any valid shell commands

> # $OH_INTERPRETER_PATH is available as the path to the OH-specific Python interpreter

Reviewer: Comments may need some clarification:
* First one says at runtime build time
* What is the comment about OH_INTERPRETER_PATH about?

> # $OH_INTERPRETER_PATH is available as the path to the OH-specific Python interpreter

Author: ```suggestion
# If you need the path to the Python interpreter in any of these commands, you can use the $OH_INTERPRETER_PATH variable
```
